### PR TITLE
Change button colors on arm and disarm

### LINKED ISSUE #94: [BUG] Arm/Disarm buttons don't give feedback
### Describe the bug

The arm/disarm buttons used to be colored red or green depending on whether we were armed/disarmed. They don't, so it's hard to tell if we're armed. Restore previous behavior.

### Related Tickets
#40

### PR DESCRIPTION
**Pull request by ShannonGriswold**
# Pull Request

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Change the colors of the arm and disarm buttons depending on whether the robot is armed or disarmed
## Related Tickets & Documents

- Related Issue #
- Closes #94

### CODE REVIEW

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]
This works great in simulation 🎉

But... I don't like how it only applies to the `Arm` widget. The `Arm` widget uses two `ButtonIndicator`s for its buttons, which inherit from our custom `IndicatorMixin` widget class. `IndicatorMixin` is also inherited by `CircleIndicator`, which made pretty green/red circles for statuses like flooding & Pi connection.

`CircleIndicator` used broken code that switched between stylesheets that are defined in `surface/gui/gui/styles` (the `.qss` files). I kinda like your solution (defining the stylesheets in code and switching between them just by setting the stylesheet) better, but I want your solution to work for everything that inherits from `IndicatorMixin`. Can you update this to work from `IndicatorMixin`?

File: src/surface/gui/gui/styles/custom_styles.py [hunk 1]
```diff
@@ -2,43 +2,35 @@
...
 
-    # A component is disabled, not expected to have any effect or perform its function because of
-    # some external factor, either another widget or something external to the gui
-    # For example, a the arm button when the pi is not connected
-    _INACTIVE = 'inactive'
-
-    # Removes any state
-    _NO_STATE = ''
+    def set_initial_stylesheet(self) -> None:
+        self._ORIGINAL_STYLESHEET = self.styleSheet()
```
Comment (benjaminwp18) [inline src/surface/gui/gui/styles/custom_styles.py:18]: SCREAMING_SNAKE is for constants, make this `self._original_stylesheet`

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]

File: src/surface/gui/gui/styles/custom_styles.py [hunk 2]
```diff
@@ -1,44 +1,71 @@
...
+        str
+            the stylesheet that corresponds to the current state of the widget
+        """
+        if self.current_state == WidgetState.ON:
+            return self._ON_STYLESHEET
+        if self.current_state == WidgetState.OFF:
+            return self._OFF_STYLESHEET
+        if self.current_state == WidgetState.INACTIVE:
+            return self._INACTIVE_STYLESHEET
+        return ''
```
Comment (benjaminwp18) [inline src/surface/gui/gui/styles/custom_styles.py:65]: Store `_ON_STYLESHEET` etc. in a dictionary of `WidgetState`s -> stylesheets

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]

File: src/surface/gui/gui/styles/custom_styles.py [hunk 3]
```diff
@@ -1,44 +1,57 @@
...
+from PyQt6.QtWidgets import QPushButton, QWidget
 
-    def set_off(self) -> None:
-        self.setProperty(IndicatorMixin._PROPERTY_NAME, IndicatorMixin._OFF)
-        self._update_style()
 
-    def set_inactive(self) -> None:
-        self.setProperty(IndicatorMixin._PROPERTY_NAME, IndicatorMixin._INACTIVE)
-        self._update_style()
+class WidgetState(Enum):
```
Comment (InvincibleRMC) [inline src/surface/gui/gui/styles/custom_styles.py:7]: use IntEnum

**Review by benjaminwp18 — APPROVED** [review]
🥳

**Pull request merged**